### DISCUSSION TITLE
Counting number of files in directories (as an option) vs. space usage

### DISCUSSION THREAD

**Discussion by ossdesign** [dundee/gdu#139] [Ideas]
There may be a better way to do this, but in doing some massive storage cleanup, I was thinking it would be nice to run GDU with a flag that told it to count the number of files in directories vs. space usage, displaying that info with same means as it currently does with space. It would of course be a niche usage case, but I know for me I keep thinking it would be nice to have.

Any thoughts on this? Any suggestions of where I should look otherwise (cross-platform as GDU is would be nice)? I am kind of new to using Github for this sort of thing, so apologies if not doing this right here.

**Comment by danie-dejager** [comment]
Would you want to identify locations consuming a lot of inodes? Looking at the count of files compared to the storage used?

**Reply by ossdesign** [reply]
I am not an expert, but I think the answer is yes. I kind of understand inodes being references to files with metadata (like a table of contents in each folder), but do not ask me details ;) I kind of assumed a lot of these "space sniffer" type programs (like GDU, which I l love btw) just added up space used calculated from similar information.

If this helped, an example would be to find folders I have that are kind of hidden but with a ton of really small files in them. The total size does not stand out, but the number of files would. Does that make sense?

If you want to know the truth, what got me thinking about this was being kind of OCD about abandoned "portable style" applications that have lots of files contained within their folder. And maybe this usage case is too niche, not something people would really care for. But thought maybe it might fit, at least was curious enough to ask about.

**Comment by dundee** [comment]
Hi @ossdesign. There is key-binding `c` for showing item count and also `C` for sorting items by item count. You will still see size of directory in the first column and the relative usage in the second column, so I think you can compare quite easily. Does this fulfil your use case?

**Reply by ossdesign** [reply]
@dundee Well, well, I tell you. I suppose if I would have dug in a little deeper, I would have realized I could already do what I was looking to get done! That is a long way of saying that yes, this seems just about right and sufficient. Thanks for letting me know. To add, I have really come to like GDU since coming across it, so many thanks for sharing this with all of us.

**Reply by dundee** [reply]
Good to hear that. Thanks!
